packages/flutter_tools/lib/src/base/context.dart

```diff
@@ -3,78 +3,39 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'package:stack_trace/stack_trace.dart';
 
 final AppContext _defaultContext = new AppContext();
 
 typedef void ErrorHandler(dynamic error, StackTrace stackTrace);
 
-/// A singleton for application functionality. This singleton can be different
-/// on a per-Zone basis.
-AppContext get context {
-  AppContext currentContext = Zone.current['context'];
-  return currentContext == null ? _defaultContext : currentContext;
-}
+AppContext _context = _defaultContext;
+
+/// A singleton for the application functionality.
+AppContext get context => _context;
 
 class AppContext {
   Map<Type, dynamic> _instances = <Type, dynamic>{};
-  Zone _zone;
-
-  bool isSet(Type type) {
```

Comment:
I think it could be simplified without removing the zone using a utility along the following lines:

```dart
/// Walks up the context chain and returns the first context containing [type].
AppContext _contextWith(Type type) {
  AppContext current = Zone.current['context'];
  while (current != null) {
    if (current._instances.containsKey(type))
      return current;
    current = _calcParent(Zone.current);
  }
  return null;
}
```

Then the method implementations can be simpler:

```dart
bool isSet(Type type) => _contextWith(type) != null;
dynamic getVariable(Type type) => _contextWith(type)?._instances?.getVariable(type);
```